WebsiteDataStoreClient::notifyBackgroundFetchChange should exit early if its delegate does not have the corresponding selector

youennf opened the pull request:
#### 7442d8810faf1007df0cf81c971645be7c647c3b
<pre>
WebsiteDataStoreClient::notifyBackgroundFetchChange should exit early if its delegate does not have the corresponding selector
<a href="https://bugs.webkit.org/show_bug.cgi?id=253432">https://bugs.webkit.org/show_bug.cgi?id=253432</a>
rdar://problem/106285707

Reviewed by Sihui Liu.

Add a boolean to store whether selector is available.
If selector is not available, bail out early.

* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:

Canonical link: <a href="https://commits.webkit.org/261365@main">https://commits.webkit.org/261365@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6427ae8035f79fa0938ff653552e544199a72a2f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111301 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20444 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43846 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/2731 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120109 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21809 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11543 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/2507 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117066 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16199 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99377 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/103950 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98152 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/31029 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44798 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12943 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/32370 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/86637 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13460 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9370 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18900 "Built successfully") | [💥 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51958 "An unexpected error occured. Recent messages:OS: Monterey (12.6.3), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7884 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15416 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (approved):
r=me on the commit for "WebsiteDataStoreClient::notifyBackgroundFetchChange should exit early if its delegate does not have the corresponding selector"

youennf committed:
WebsiteDataStoreClient::notifyBackgroundFetchChange should exit early if its delegate does not have the corresponding selector
https://bugs.webkit.org/show_bug.cgi?id=253432
rdar://problem/106285707

Reviewed by Sihui Liu.

Add a boolean to store whether selector is available.
If selector is not available, bail out early.

* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:

Canonical link: https://commits.webkit.org/261365@main
